src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -28,6 +29,11 @@ public QueryTimeouts(IDocumentStore store)
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
+            if(abort)
+            {
+                return new TimeoutsChunk(new TimeoutsChunk.Timeout[ 0 ], DateTime.UtcNow.AddHours(1));
```

Comment:
I mean moving the abort method to after the raven query runs. The query could be running while the Shutdown() method is called.